contracts/src/token/erc1155/mod.rs

```diff
@@ -0,0 +1,1973 @@
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `from` - Account to burn tokens from.
+    /// * `ids` - Array of all token ids to be burnt.
+    /// * `values` - Array of all amount of tokens to be burnt.
+    ///
+    /// # Errors
+    ///
+    /// If `from` is the `Address::ZERO`, then the error
+    /// [`Error::InvalidSender`] is returned.
+    /// If length of `ids` is not equal to length of `values`, then the
+    /// error [`Error::InvalidArrayLength`] is returned.
+    /// If `value` is greater than the balance of the `from` account,
+    /// then the error [`Error::InsufficientBalance`] is returned.
+    ///
+    /// # Events
+    ///
+    /// Emits a [`TransferSingle`] event if the arrays contain one element, and
+    /// [`TransferBatch`] otherwise.
+    ///
+    /// # Panics
+    ///
+    /// Should not panic.
+    fn _do_burn(
+        &mut self,
+        from: Address,
+        ids: Vec<U256>,
+        values: Vec<U256>,
+    ) -> Result<(), Error> {
+        if from.is_zero() {
+            return Err(Error::InvalidSender(ERC1155InvalidSender {
+                sender: from,
+            }));
+        }
+        self._update_with_acceptance_check(
+            from,
+            Address::ZERO,
+            ids,
+            values,
+            &vec![].into(),
+        )?;
+        Ok(())
+    }
+
+    /// Transfers `values` of tokens specified by `ids` from `from` to `to`.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `from` - Account to transfer tokens from.
+    /// * `to` - Account of the recipient.
+    /// * `ids` - Array of all token ids.
+    /// * `values` - Array of all amount of tokens to be transferred.
+    /// * `data` - Additional data with no specified format, sent in call to
+    ///   `to`.
+    ///
+    /// # Errors
+    ///
+    /// If `to` is the `Address::ZERO`, then the error
+    /// [`Error::InvalidReceiver`] is returned.
+    /// If `from` is the `Address::ZERO`, then the error
+    /// [`Error::InvalidSender`] is returned.
+    /// If length of `ids` is not equal to length of `values`, then the
+    /// error [`Error::InvalidArrayLength`] is returned.
+    /// If `value` is greater than the balance of the `from` account,
+    /// then the error [`Error::InsufficientBalance`] is returned.
+    /// If [`IERC1155Receiver::on_erc_1155_received`] hasn't returned its
+    /// interface id or returned with error, then the error
+    /// [`Error::InvalidReceiver`] is returned.
+    /// If [`IERC1155Receiver::on_erc_1155_batch_received`] hasn't returned its
+    /// interface id or returned with error, then the error
+    /// [`Error::InvalidReceiver`] is returned.
+    ///
+    /// # Events
+    ///
+    /// Emits a [`TransferSingle`] event if the arrays contain one element, and
+    /// [`TransferBatch`] otherwise.
+    ///
+    /// # Panics
+    ///
+    /// Should not panic.
```

Comment:
```suggestion
    /// If updated balance exceeds `U256::MAX`.
```
Increasing `to`'s balance may cause an overflow, see [do_update](https://github.com/OpenZeppelin/rust-contracts-stylus/pull/275/files#diff-00143f9b7a054f52edef59419a37d3c7939709acad52350c102ffffff3809342R1038)